src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1314,8 +1337,1089 @@ public boolean mapSwitchControlValue(String value) {
+				createNumeric(stats, streamGroup + StreamControllingMetric.SOURCE_PORT.getName(), checkForNullData(streamInfo.getSourcePort().toString())));
+
+		// Populate port numeric control
+		addAdvanceControlProperties(advancedControllableProperties,
+				createNumeric(stats, streamGroup + StreamControllingMetric.DESTINATION_PORT.getName(), checkForNullData(streamInfo.getPort().toString())));
+
+		// Populate latency numeric control
+		addAdvanceControlProperties(advancedControllableProperties,
+				createNumeric(stats, streamGroup + StreamControllingMetric.LATENCY.getName(), checkForNullData(streamInfo.getLatency().toString())));
+	}
+
+	/**
+	 * This method is used for populate all stream control properties :
+	 * Protocol: TS over SRT
+	 * SRT mode: Rendezvous
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param streamInfo stream config info
+	 */
+	private void populateStreamControlCaseTSOverSRTRendezvous(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, StreamInfo streamInfo, String streamGroup) {
+		// Populate Address text control
+		if (streamInfo.getAddress().equals(DecoderConstant.ADDRESS_ANY)) {
+			streamInfo.setAddress(DecoderConstant.EMPTY);
+		}
+		addAdvanceControlProperties(advancedControllableProperties,
+				createText(stats, streamGroup + StreamControllingMetric.ADDRESS.getName(), streamInfo.getAddress()));
+
+		// Populate source port (auto assign)
+		stats.put(streamGroup + StreamControllingMetric.SOURCE_PORT.getName(), streamInfo.getPort().toString());
+
+		// Populate port numeric control
+		addAdvanceControlProperties(advancedControllableProperties,
+				createNumeric(stats, streamGroup + StreamControllingMetric.DESTINATION_PORT.getName(), checkForNullData(streamInfo.getPort().toString())));
+
+		// Populate latency numeric control
+		addAdvanceControlProperties(advancedControllableProperties,
+				createNumeric(stats, streamGroup + StreamControllingMetric.LATENCY.getName(), checkForNullData(streamInfo.getLatency().toString())));
+	}
+
+	/**
+	 * This method is used for populate apply change button and cancel button of stream control:
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param streamID ID of stream
+	 */
+	private void populateApplyChangeAndCancelButtonForStream(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, Integer streamID) {
+		Optional<StreamInfo> streamInfoDTO = this.streamInfoDTOList.stream().filter(st -> streamID.equals(st.getId())).findFirst();
+		Optional<StreamInfo> streamInfo = this.localStreamInfoList.stream().filter(st -> streamID.equals(st.getId())).findFirst();
+		Encapsulation encapsulation = streamInfoDTO.get().getEncapsulation();
```

Comment:
is it safe to use optionals this way? Would it be better doing something like
streamInfoDTO.transform(streamInfo.getEncapsulation()).orNull() or at least checking the presence with "isPresent()" before?

Please check other optional usages as well

Reply:
I fixed it.